### PR TITLE
Rename ItcProps to ItcGraphQuerier, fix the reusability

### PR DESCRIPTION
**Pull request by toddburnside**

### CODE REVIEW

File: explore/src/main/scala/explore/itc/ItcGraphQueryer.scala [hunk 1]
```diff
@@ -98,10 +98,16 @@ case class ItcProps(
 
   val targets: List[ItcTarget] = itcTargets.foldMap(_.toList)
 
-  private val queryProps: List[Option[?]] =
-    List(itcTargets, finalConfig, wavelength, instrumentConfig, signalToNoise)
+  private val queryProps =
+    (wavelength,
+     signalToNoise,
+     signalToNoiseAt,
+     constraints.some,
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/itc/ItcGraphQueryer.scala:105]: I don't think `constraints` is needed. All that we need from it is already factored into `instrumentConfig`.

**Review by rpiaggio — APPROVED** [review]
LGTM

File: explore/src/main/scala/explore/itc/ItcGraphQuerier.scala [hunk 2]
```diff
@@ -98,10 +98,16 @@ case class ItcProps(
...
   val targets: List[ItcTarget] = itcTargets.foldMap(_.toList)
 
-  private val queryProps: List[Option[?]] =
-    List(itcTargets, finalConfig, wavelength, instrumentConfig, signalToNoise)
+  private val queryProps =
+    (wavelength,
+     signalToNoise,
+     signalToNoiseAt,
+     constraints.some,
+     itcTargets,
```
Comment (cquiroz) [inline explore/src/main/scala/explore/itc/ItcGraphQuerier.scala:106]: Is it nott enough to query only for target ids?

Comment (toddburnside) [reply explore/src/main/scala/explore/itc/ItcGraphQuerier.scala:106]: Does the itc know the target ids?

Comment (toddburnside) [reply explore/src/main/scala/explore/itc/ItcGraphQuerier.scala:106]: This would be a bigger change. I'm going to merge, but we can continue discussing.

Comment (cquiroz) [reply explore/src/main/scala/explore/itc/ItcGraphQuerier.scala:106]: If I recall this is used only to tell  if the properties have changed, and using the ids would bee more efficient that comparing the whole target

Comment (toddburnside) [reply explore/src/main/scala/explore/itc/ItcGraphQuerier.scala:106]: The ItcTargets, with the gaia bands removed, are passed to `ItcClient.spectroscopyIntegrationTimeAndGraphs`. So, I assume the itc needs the target information?

Comment (cquiroz) [reply explore/src/main/scala/explore/itc/ItcGraphQuerier.scala:106]: I see, for reusability would the `id` be enough?

Comment (toddburnside) [reply explore/src/main/scala/explore/itc/ItcGraphQuerier.scala:106]: I don't think so. If a target is edited, it keeps the same id but the itc results could be affected.

**Pull request merged**